backend/shared/shared/oidc/views/eauth_views.py

```diff
@@ -56,13 +61,37 @@ def register_user(self, person_id):
         return response.json()
 
     def get(self, request):
-        """Eauth client authentication initialization HTTP endpoint"""
-        if not (request.session.get("oidc_access_token")):
+        """Eauth client authentication initialization HTTP endpoint
+
+        NOTE: We should avoid raising exceptions from the method, because it results in user's auth flow
+        ending on Django's 500 error page. We should instead call `self.login_failure()` to redirect the
+        user to the login error page in the UI.
+        """
+        oidc_access_token = request.session.get("oidc_access_token")
+        if not oidc_access_token:
             return self.login_failure()
 
         user_info = get_userinfo(request)
 
+        # When authenticating via Tunnistus service, we can read `user_ssn` from `user_info`,
+        # but for authentication via Tunnistamo, we need to call Helsinki Profile GraphQL API.
         user_ssn = user_info.get("national_id_num")
+        if user_ssn is None:
+            try:
+                profile = HelsinkiProfileClient().get_profile(oidc_access_token)
+            except HelsinkiProfileException as e:
+                logger.warning(
```

Comment:
One option here would be to use `logger.exception(...)` [which should automatically](https://docs.python.org/3/library/logging.html#logging.Logger.exception) put the exception info to the logging message. It will be an logged at ERROR level, which is maybe something that we don't want to do here. Not saying that this needs to be changed. Just pointing out that there's this special logger which can be used inside an exception handler.